app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -132,17 +137,12 @@ public void onDrawerStateChanged(int newState) {
         }
 
         checkInformedConsent();
+    }
 
-        // XXX: This is probably not correct: we would like to send
-        // info to the orchestrator only when the network or any other
-        // orchestrator parameter like country code changed.
-        //TODO remove this code in 1.1.6
-        String token = FirebaseInstanceId.getInstance().getToken();
-        if (token != null) {
-            NotificationService ns = NotificationService.getInstance(this);
-            ns.setDevice_token(token);
-            ns.sendRegistrationToServer();
-        }
+    @Override
+    public void onResume() {
+        super.onResume();
+        TestData.getInstance(this, this).addObserver(this);
```

Comment:
So, I've read the code of `getInstance` and `addObserver`. Now, I also get why this is called `TestData` :-).

What is not totally clear to me is the logic of `getInstance`: can you explain it? (I believe this could also be useful knowledge to me for refactoring the tests to use intents so to fully decouple running tests and view.)

Reply:
I need which class is "active" to manage the Observables

Reply:
I understand. I think that's not super obvious from the API. As I said, I find the fact that sometimes a singleton is used and sometimes static methods a bit strange.